pandas/io/json/_json.py

```diff
@@ -1246,6 +1248,14 @@ def _parse(self) -> None:
                 for k, v in loads(json, precise_float=self.precise_float).items()
             }
             self.check_keys_split(decoded)
+            orig_names = [
+                (tuple(col) if isinstance(col, list) else col)
+                for col in decoded["columns"]
+            ]
```

Comment:
currently, this isn't necessary, because the test is xfailed, so it will pass even if there's an exception thrown (which I is what this line was meant to address?)

I'm not too keen on adding a line of code which doesn't have any effect on the test suite (someone else might accidentally remove it as part of a refactor, and the test suite would still pass), so how about either:
- remove this, and just letting the code raise: as your example shows, multiindex with `columns='split'` wasn't supported to begin with, so it's OK for it to throw an error;
- OR keep this, but change the test to check that the result value has tuples as column names, with a comment saying that the return type should be changed (instead of xfailing)

Reply:
Good point. It's a bit more complex. This line does have an effect, and will make other tests fail if removed. A multiindex is represented internally in some places as a list of tuples. But JSON doesn't have the concept of tuples, so in the JSON it's represented as a list of lists. When calling dedup columns and the dataframe constructor, providing a list of lists is interpreted as a normal index with lists as values, which is a non-hashable type, and it raises a TypeError. Making them tuples will convert the values to the multiindex, and will make the JSON be loaded with a multiindex.

The part that is still missing in this PR is that the loaded multiindex is not the same as the original one, if the JSON was saves with `to_json`. That's what the other issue will fix, and what the xfailed test is checking.

What it makes sense based on your comment is to be more explicit on when the test should xfail. I added that the xfail needs to be for `AssertionError` (the compared values are different). So now, if this line is removed, the test will fail, as the error for trying to create an index with a list of lists is a `TypeError`, and the test will check that the code doesn't raise, and will need that a value is returned, just different from the expected one.

Does this make sense to you? I wouldn't implement anything more complex than this here, since fixing the other bug shouldn't be difficult. We can implement the other PR first too, but probably not worth it.

Reply:
looks good, thanks!